flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/declarative/StateTrackingMockExecutionGraph.java

```diff
@@ -0,0 +1,459 @@
+        return true;
+    }
+
+    @Override
+    public JobStatus getState() {
+        return state;
+    }
+
+    @Override
+    public CompletableFuture<JobStatus> getTerminationFuture() {
+        return terminationFuture;
+    }
+
+    @Override
+    public Executor getFutureExecutor() {
+        return ForkJoinPool.commonPool();
+    }
+
+    @Override
+    public void registerExecution(Execution exec) {
+        // no-op
+    }
+
+    @Override
+    public void cancel() {
+        state = JobStatus.CANCELLING;
+    }
+
+    @Override
+    public void failJob(Throwable cause) {
+        state = JobStatus.FAILING;
+    }
+
+    @Override
+    public void suspend(Throwable suspensionCause) {
+        state = JobStatus.SUSPENDED;
+    }
+
+    @Override
+    public void transitionToRunning() {
+        state = JobStatus.RUNNING;
+    }
+
+    // --- interface implementations: methods for creating an archived execution graph
+
+    @Override
+    public int getTotalNumberOfVertices() {
+        return 0;
+    }
+
+    @Override
+    public Iterable<ExecutionJobVertex> getVerticesTopologically() {
+        return Collections.emptyList();
+    }
+
+    @Override
+    public Map<JobVertexID, ExecutionJobVertex> getAllVertices() {
+        return Collections.emptyMap();
+    }
+
+    @Override
+    public Map<String, SerializedValue<OptionalFailure<Object>>> getAccumulatorsSerialized() {
+        return Collections.emptyMap();
+    }
+
+    @Override
+    public long getStatusTimestamp(JobStatus status) {
+        return 0L;
+    }
+
+    @Override
+    public String getJsonPlan() {
+        return "";
+    }
+
+    @Override
+    public JobID getJobID() {
+        return jobId;
+    }
+
+    @Override
+    public String getJobName() {
+        return "testJob";
+    }
+
+    @Nullable
+    @Override
+    public ErrorInfo getFailureInfo() {
+        return null;
+    }
+
+    @Nullable
+    @Override
+    public ArchivedExecutionConfig getArchivedExecutionConfig() {
+        return archivedExecutionConfig;
+    }
+
+    @Override
+    public boolean isStoppable() {
+        return false;
+    }
+
+    @Nullable
+    @Override
+    public CheckpointCoordinatorConfiguration getCheckpointCoordinatorConfiguration() {
+        return null;
+    }
+
+    @Nullable
+    @Override
+    public CheckpointStatsSnapshot getCheckpointStatsSnapshot() {
+        return null;
+    }
+
+    @Override
+    public Optional<String> getStateBackendName() {
+        return Optional.empty();
+    }
+
+    @Override
+    public Optional<String> getCheckpointStorageName() {
+        return Optional.empty();
+    }
+
+    @Override
+    public StringifiedAccumulatorResult[] getAccumulatorResultsStringified() {
+        return new StringifiedAccumulatorResult[0];
+    }
+
+    // -- remaining interface implementations: all unsupported
+
+    @Override
+    public boolean isArchived() {
+        throw new UnsupportedOperationException();
```

Comment:
we could remove this method entirely

Reply:
You mean adding a default method returning false in the `ExecutionGraph` interface? or just always returning false in the mock?

Reply:
I mean to literally delete the `isArchived` method. It is not used in a meaningful way.

Reply:
Ah, good catch. Will remove it.